### DISCUSSION TITLE
Why is the default install Microsoft store?

### DISCUSSION THREAD

**Discussion by illspirit** [stoatchat/discussions#615] [Q&A] [answered]
I thought this app was for privacy focused individuals? I understand the .exes can be found on Github but I'm just wondering why Microsoft store is involved at all? I have most of the store functionality removed and MS servers blocked via DNS.

**Comment by Rexogamer** [comment, marked as answer]
The Microsoft Store is easy to use for a lot of users (auto-updates, easy access/app management etc) and is entirely optional. Our main aim is to provide Revolt as widely as possible, and this helps with that - in addition, it's basically identical to the "normal" builds, and does not include/lack anything compared to them.

You are more than welcome to use the executable files or Winget and block connections to Microsoft's servers - the MS Store is provided as an entirely optional method of installing the app, and I certainly think having it as an option for those who want/need it while not impacting those who don't is valuable.

**Reply by EdgelordUltima** [reply]
The key point is that you made Microsoft Store the big, default, user-friendly option.
While downloading an executable requires going here, to github - the scary, scary place where people (*gasp*) post code.

![Revolt-microsoft-store-default-option1](https://github.com/revoltchat/revolt/assets/145729512/5590a0d4-46c4-4040-ac2a-6f3742c9049f).

You may not be a "giant evil megacorp", but you certainly don't live up to your marketing.
You're not revolting against anything without featuring E2E in user servers. That's just a fact.
Most normies will keep using discord, and most people who want privacy can't trust or recommend you.

**Reply by Rexogamer** [reply]
I agree with [what Lea said below](https://github.com/orgs/revoltchat/discussions/615#discussioncomment-6934006) - if you're this worried about privacy, why use an operating system that has been widely criticised for its data collection and related practises?

**Reply by EdgelordUltima** [reply]
It's false dichotomy. "Either you fully obssess with anonymity, or you give up on all security".
Everyone wants privacy, but there's a spectrum of how much hassle a person is willing to go through for it.

Those willing to install Linux will always be an extreme minority - but that's besides the point, which is that, there are people willing to switch from Discord to a more privacy-oriented platform, and they are your primary audience, yet you fail to actually provide any privacy.

I'm not obsessed with privacy. I asked my friend on twitter for his discord, and since he's cautious, he introduced me to Element.
Your product is more user-friendly than Element, but that's not enough for me to be able to recommend Revolt to that friend, because it's very clear that you guys care about privacy much less than your competitor, or else you would have long added E2E to DMs **and** servers.

**Reply by insertish** [reply]
I just want to comment on a few things:
- I don't see how the default option on the website is relevant to any privacy-conscious individual. For those people, I imagine they would instinctively seek out and check if there is an alternative way to download the app if they do not wish to use the Microsoft Store. (I don't see a problem with this.)
- Our audience is not exclusive to privacy-oriented individuals, we are catering to a general audience, and distributing through the Microsoft Store offers the best end-user experience. (I also don't think you can assert that privacy-oriented individuals are our primary audience. This is not to say that it's not a core principle or anything but it's not something we market using either, because like you've said, we don't have e2ee yet so I feel it's misleading to promote the platform based on that hence we don't.)
- If you want an explanation of why e2ee isn't here yet, it's just because we're a very small team, other things like usability and basic feature parity (with other platforms) is taking priority right now. It is however, slated to come after voice/video calling on the roadmap.

**Reply by EdgelordUltima** [reply]
I appreciate your response.
- I find it hard to believe that "general audiece" prefers using Microsoft Store over downloading an executable from a website.
Github scares them away, and Microsoft Store is additional service required, and is thus an extra hassle.
A few clicks to download an executable from a website's home page is much easier and attractive to a general audience, hence my confusion.

- My issue isn't so much with e2e not being implemented in DMs yet, as much as with the fact that user servers aren't mentioned. 
Why are e2e plans limited to DMs and Group Chats? Will it eventually be added to user servers?

If that is the case, then I sincerely hope to see that in the future, whenever that may be.

**Reply by IAmTomahawkx** [reply]
> A few clicks to download an executable from a website's home page is much easier and attractive to a general audience, hence my confusion.

According to whom? If there's actual data to back that up I'd be interested to see it. We believe that the MS store provides an easy way for "non-techy" people to get access to the app, and it provides them with built-in tools like auto update and app management (like rexo said originally). The Microsoft store is already part of windows, we're not asking you to download third party app stores (or use the store at all). Users who don't care can simply click a button and get the app in the easiest to use form factor. More advanced users are absolutely free to download binaries directly off of GitHub. 

> Why are e2e plans limited to DMs and Group Chats? Will it eventually be added to user servers?

Scaling encryption is Hard(tm). Most services limit the size of an encrypted chat to sub-one-thousand users. We've not put servers on the encryption roadmap as we have not figured out a way to do it in a way that satisfies our needs. That being said we haven't put many resources into it yet, as we've not been focusing on encryption. It may end up on the roadmap once we start putting more effort into encryption and find a good way to do it.

**Reply by hubdot** [reply]
The Microsoft store is an extremely bad place for "non techy" OR "techy" people to download apps from.

In fact if you search for Revolt, or revolt chat, on the Microsoft store, you don't even get the app suggested on the MS store!
It shows me TeamSpeak, eM Client or even ChatGPT as results - but NO revolt.

Not only does it require a number of frameworks to function perfectly (for many people the download link doesn't even open the MS store...), it also requires a fully signed in Windows, among other things.

Discord can do it
Guilded can
Element can
Slack can
Teams does it
Mumble can
TeamSpeak...
heck even Trillian, mIRC...

It's really not making any sense to argument against common practice here. It only hurts the users.

I tried to adopt a number of DC users to revolt and every single one complained. Only a fraction ever tried it via github and so it fell flat.

Not even my grand parents use the Microsoft store, nobody I know uses it. That's why Windows phones failed and never got off the ground.

TL/DR
This is factually a bad approach.
No popular or relevant communication software aside from Revolt uses this approach.
You can't search for it on the MS store. It doesn't show up in the results.
The DL link to the MS is often bugged. MS store has too many dependencies to work correctly at all.
If it was an option for a select niche of MS store fans okay - but this is not helpful.

You should seriously re-consider. I'm tired of being tasked to berate people that it's actually not a bad app and just has questionable decision making behind it's release channels.

**Reply by illspirit** [reply]
They won't consider. Clown dev team.

**Reply by DeclanChidlow** [reply]
Thanks for the in-depth message @hubdot. Definitely some points there that should be considered and that might require re-evaluation.

@illspirit making snide remarks about Revolt's volunteer development team isn't kind, and doesn't have a place in this community. Please don't do it again.

**Comment by illspirit** [comment]
I thought the idea behind the app was privacy. It's a branding and image thing. I guess I thought this was a privacy oriented alternative to Discord. Reading through the website I can see it's definitely focused on data collection like all the other big ones.

E2E isn't even a thing for DMs?

What is the meaning behind the name? What are you Revolting against? It's certainly not against big corporations.

**Reply by Rexogamer** [reply]
Hostility aside:
- we do not collect any extraneous data and aside from one of the ***options*** for installing the app being an app store you dislike I don't see what gives you this impression
- not yet, but we do not claim otherwise anywhere and very much intend to work on it after other features/rewrites are done
- again, just because one of the ***options*** for downloading Revolt is an app store owned by Microsoft doesn't mean we're suddenly Evil Spying Big Tech Corporation No. 238 - we simply want to make the app available to more people by distributing it via stores they use.

**Reply by sussycatgirl** [reply]
In addition to this, you wouldn't be seeing the  Microsoft Store as default option if you weren't using Windows in the first place. I think it's a bit hypocritical to get mad as us for offering it as an option while using the Microsoft products you despise so much yourself.
